pageserver/src/page_service.rs

```diff
@@ -285,6 +299,8 @@ struct PageServerHandler {
     /// For each query received over the connection,
     /// `process_query` creates a child context from this one.
     connection_ctx: RequestContext,
+
+    shard_timelines: HashMap<ShardIndex, HandlerTimeline>,
```

Comment:
Looks like `PageServerHandler` has the same lifetime as the connection it serves. How long lived are these connections generally? I'm wondering about the total memory usage of these maps since we don't delete from them.

Reply:
They're pretty long lived, but the size of the cache is almost always tiny https://github.com/neondatabase/neon/pull/6037#discussion_r1449210551

I've added a comment in 90fd1f950 to make this clearer, as you and Christian both raised the point (which is good, we should always look on in-memory caches with suspicious eyes.)